packages/datepickers/src/examples/default-usage.md

```diff
@@ -0,0 +1,27 @@
+The `<Datepicker>` takes a native `Date()` object as its input. It is able to
+accept several varieties of user input to parse it's date.
+
+By default it allows the standard `Date.parse()` formats, including:
+
+- `09/04/1986`
+- `Sep 4, 1986`
+- `September 4th, 1986`
+
+These values are localized and will change their format based on the provided locale.
+This logic may be customized with the `customParseDate` prop to allow more advanced
+date parsing with [momentjs](https://momentjs.com/) or [date-fns](https://date-fns.org/).
+
+```jsx
+const { Field, Label, Input } = require('@zendeskgarden/react-forms/src');
+
+initialState = {
+  value: new Date()
+};
+
+<Field>
+  <Label>Select a date</Label>
+  <Datepicker value={state.value} onChange={newDate => setState({ value: newDate })}>
```

Comment:
> Clicking on a datepicker field should select the entry text for me.

I discussed this with @ginnywood during the implementation and we decided to preserve the "standard" input selection. This allows users to click into a Datepicker and immediately start editing a specific value. If the text was selected on click it would take several keyboard presses to perform an edit.

Example:
![2019-07-03 10-31-09 2019-07-03 10_31_37](https://user-images.githubusercontent.com/4030377/60612569-c4382f00-9d7d-11e9-9cc3-a6a239589865.gif)

Reply:
Also, if you click the label of the it will select the full-text. Same as a native form control.

Reply:
> Date entry with no year defaults to 2001. I'd want it to fallback to current year.

This was a side-effect of using the `Date()` parse method. It is unique per browser (Safari defaults to 2000 for example). Since we're already using `date-fns` I've moved this logic to their `parse()` method and am allowing the [P, PP, PPP formats for entry](https://date-fns.org/v2.0.0-beta.2/docs/parse). This should include some additional flexibility.